packages/clerk-js/src/ui/signUp/util.ts

```diff
@@ -0,0 +1,27 @@
+import { SignUpResource } from '@clerk/types';
+
+type CompleteSignUpFlowProps = {
+  signUp: SignUpResource;
+  signUpUrl?: string;
+  navigate: (to: string) => Promise<void>;
+  handleComplete?: () => Promise<void>;
+};
+
+export const CompleteSignUpFlow = ({
+  signUp,
+  signUpUrl,
+  navigate,
+  handleComplete,
+}: CompleteSignUpFlowProps): Promise<void> | undefined => {
+  if (signUp.status === 'complete') {
+    return handleComplete && handleComplete();
+  } else if (signUp.status === 'missing_requirements') {
+    if (signUp.unverifiedFields?.includes('email_address')) {
+      return navigate(signUpUrl ? signUpUrl + '/verify-email-address' : 'verify-email-address');
+    }
+    if (signUp.unverifiedFields?.includes('phone_number')) {
+      return navigate(signUpUrl ? signUpUrl + '/verify-phone-number' : 'verify-phone-number');
+    }
+  }
```

Comment:
❓ 
If we pass the signUpUrl from contexes like `useCoreSignUp` , will the interpolation work as expected ?
Reminder these URLs go through https://github.com/clerkinc/javascript/blob/main/packages/clerk-js/src/ui/contexts/ClerkUIComponentsContext.tsx#L60-L67

cc: @SokratisVidros @nikosdouvlis

Reply:
@igneel64 it should work as expected if we use the `buildURL` function and not just concatenate strings. Up until now, no concatenation was required as we were just doing `navigate('verify-email-address');`.